core/txindexer.go

```diff
@@ -145,7 +136,12 @@ func (indexer *txIndexer) loop(chain *BlockChain) {
 		done = make(chan struct{})
 		lastHead = head.Number.Uint64()
 		indexer.chain.wg.Add(1)
-		go indexer.run(rawdb.ReadTxIndexTail(indexer.db), head.Number.Uint64(), stop, done)
+		go func() {
```

Comment:
Am I correct in understanding that this (and the similar go routine below) is where the fix occurs, because the call to `rawdb.ReadTxIndexTail` is now inside the lock?

Does `ResetToStateSyncedBlock` honour the same lock?

Reply:
yes, `ResetToStateSyncedBlock` uses `repairTxIndexTail` which holds the same lock